packages/eui/src/components/code/code_block.test.tsx

```diff
@@ -46,6 +46,19 @@ describe('EuiCodeBlock', () => {
       });
     });
 
+    describe('copyAriaLabel', () => {
+      it('is rendered', () => {
+        const customLabel = 'Copy this code';
+        const { container } = render(
+          <EuiCodeBlock isCopyable copyAriaLabel={customLabel}>
+            {code}
+          </EuiCodeBlock>
+        );
+
+        expect(container.firstChild).toMatchSnapshot();
```

Comment:
I generally prefer specific test assertions as opposed to snapshots.  As we've found time and time again, devs tend not to look at snapshots closely, so things tend to get missed and "pass" CI when they should get caught. I think my favorite snapshot I've seen so far in Kibana is just a snapshot of `null` 😂 

For historical context: this is something that I've tried to slowly move the EUI codebase towards, but unfortunately there's a lot of legacy snapshot tests still all over the place mostly because we have so many dang components 😅 Also, I should mention I do like a single `it('renders')` first test emitting a snapshot, but IMO prop tests should have specific goals in mind and generally be able to verify those goals with targeted assertions.

This test in particular should be very easy to target via RTL selectors instead of a snapshot, to accomplish what we're looking for:

```suggestion
        const { getByTestSubject } = render(
          <EuiCodeBlock isCopyable copyAriaLabel={customLabel}>
            {code}
          </EuiCodeBlock>
        );

        expect(getByTestSubject('euiCodeBlockCopy')).toHaveAttribute('aria-label', customLabel);
```

Reply:
This is an incredibly valuable knowledge drop ☝🏻 At first, I used a similar assertion as you're proposing. But then seeing all test cases leveraging snapshots, I went with consistency.

It seems you have a lot of ideas for improvements and I appreciate that you're sharing them! 💚 What better place to pay off some tech debt then on a semi-related PR?

How about we sit down and write the tech debt that you had planned / started working on (preferably in form of issues)? Yeah, many fear planning these sort of things because the tech-debt tends to get lost somewhere at the bottom of the backlog. But I'll gladly take over those chores cause it's easy to wrap your head around, it's a repeatable task that lets you explore the codebase a bit more. This is something I could pick up each Friday.

Reply:
Yes sorry I forgot to clarify, no need to update the existing tests in this PR (unless you strongly feel like it!) - I just want to make sure any new tests we added are better than our old practices!